ext/ffi/lib.rs

```diff
@@ -133,9 +125,11 @@ fn event_loop_middleware(
   // FFI callbacks coming in from other threads will call in and get queued.
   let mut maybe_scheduling = false;
```

Comment:
This should be a slightly more efficient way to do it -- we can avoid the borrow by returning a value from the inner loop. Note that we should also probably have a SmallVec here (put a TODO in there for this):

```rust
  // FFI callbacks coming in from other threads will call in and get queued.
  let mut maybe_scheduling = false;

  let work_items = {
    let mut op_state = op_state_rc.borrow_mut();
    if let Some(ffi_state) = op_state.try_borrow_mut::<FfiState>() {
      // TODO(mmastrac): This should be a SmallVec to avoid allocations in most cases
      let mut work_items = Vec::with_capacity(1);

      while let Ok(Some(async_work_fut)) =
        ffi_state.async_work_receiver.try_next()
      {
        // Move received items to a temporary vector so that we can drop the `op_state` borrow before we do the work.
        work_items.push(async_work_fut);
        maybe_scheduling = true;
      }
  
      Some(work_items)
    } else {
      None
    }
  };

  if let Some(work_items) = work_items {
    for async_work_fut in work_items.into_iter() {
      async_work_fut();
    }
  }

  maybe_scheduling
```